File: NewsApp/app/src/main/java/com/example/shakil/newsapp/Interface/NewsService.java
package com.example.shakil.newsapp.Interface;

import com.example.shakil.newsapp.Common.Common;
import com.example.shakil.newsapp.Model.News;
import com.example.shakil.newsapp.Model.WebSite;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Url;

public interface NewsService {
    @GET("v2/sources?apiKey=" + Common.API_KEY)
    Call<WebSite> getSources();

    @GET
    Call<News> getNewestArticles(@Url String url);
}

File: KotlinNewsApp/app/src/main/java/com/example/shakil/kotlinnewsapp/Interface/NewsService.kt
package com.example.shakil.kotlinnewsapp.Interface

import com.example.shakil.kotlinnewsapp.Model.News
import com.example.shakil.kotlinnewsapp.Model.WebSite
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Url

interface NewsService {
    @get: GET("v2/sources?apiKey=7cc6f9c8dfa44fe993467007ada10a9e")
    val sources: Call<WebSite>

    @GET
    fun getNewsFromSource(@Url url: String): Call<News>
}